### ISSUE TITLE
Improve accessibility of the translation editor for screen reader and keyboard users

### ISSUE THREAD

**Issue by mathjazz** [mozilla/pontoon#4290]
Forwarding a message from the #l10n channel:

_Hi all! I've been communicating with a Pontoon user recently who has noted some accessibility issues with the interface. Particularly, they wrote the following to be forwarded to any relevant parties_

_As a screen reader user, I have encountered several challenges while using Pontoon. Some of the difficulties include unlabeled buttons and controls, uncertainty about whether translations have been saved successfully, difficulty navigating between resources and strings using only the keyboard, and limited feedback from the screen reader in certain parts of the interface. In addition, it is often difficult to know which translation string currently has keyboard focus, whether an action has been completed successfully, or whether I have moved to the next string after submitting a translation. These challenges make the translation process slower and more difficult than it should be. Any improvements in these areas would make Pontoon much more accessible for blind translators and would encourage greater participation from people with visual impairments._
